test/test_live_coinc_compare.py

```diff
@@ -0,0 +1,141 @@
+"""Mock simulation to easily test and profile PyCBC Live's coincidence code."""
+
+import unittest
+from types import SimpleNamespace
+import numpy as np
+# Some duplicate imports, but I want to copy code without changing it!
+import numpy, logging, pycbc.pnutils, pycbc.conversions, copy, lal
+import cProfile
+from astropy.utils.data import download_file
+from pycbc import gps_now
+from pycbc.events.coinc import LiveCoincTimeslideBackgroundEstimator as Coincer
+import pycbc.events.coinc
+from utils import simple_exit
+import validation_code.old_coinc as old_coinc
+
+OriginalCoincer = old_coinc.LiveCoincTimeslideBackgroundEstimator
+
+class SingleDetTrigSimulator:
+    """An object that simulates single-detector triggers in the same format
+    as produced by the matched-filtering processes of PyCBC Live.
+    """
+    def __init__(self, num_templates, analysis_chunk, detectors, num_trigs_per_block):
+        self.num_templates = num_templates
+        self.detectors = detectors
+        self.analysis_chunk = analysis_chunk
+        self.start_time = gps_now()
+        self.num_trigs = num_trigs_per_block
+
+    def get_trigs(self):
+        trigs = {}
+        for det in self.detectors:
+            trigs[det] = {
+                "snr": np.random.uniform(4.5, 10, size=self.num_trigs).astype(np.float32),
+                "end_time": np.random.uniform(
+                    self.start_time,
+                    self.start_time + self.analysis_chunk,
+                    size=self.num_trigs
+                ).astype(np.float64),
+                "chisq": np.random.uniform(0.5, 1.5, size=self.num_trigs).astype(np.float32),
+                "chisq_dof": np.ones(self.num_trigs, dtype=np.int32) * 10,
+                "coa_phase": np.random.uniform(0, 2*np.pi, size=self.num_trigs).astype(np.float32),
+                "sigmasq": np.ones(self.num_trigs, dtype=np.float32),  # FIXME (maybe)
+                "template_id": np.random.uniform(
+                    0,
+                    self.num_templates,
+                    size=self.num_trigs
+                ).astype(np.int32)
+            }
+        self.start_time += self.analysis_chunk
+        return trigs
+
+
+class TestPyCBCLiveCoinc(unittest.TestCase):
+    def setUp(self, *args):
+        # Uncomment for more verbosity
+        # logging.basicConfig(format="%(asctime)s %(message)s",
+        #                     level=logging.INFO)
+
+        # simulate the `args` object we normally get from the command line arguments
+
+        url = 'https://github.com/gwastro/pycbc-config/raw/master/'
+        url += 'test_data_files/{}-PTA_HISTOGRAM.hdf'
+        stat_file_paths = [
+            download_file(url.format("H1L1"), cache=True),
+            download_file(url.format("H1V1"), cache=True),
+            download_file(url.format("L1V1"), cache=True)
```

Comment:
Is the test not testing just HL? Are HV and LV really needed?

Reply:
Yeah, I guess we don't need the others. I can fix this.